src/extension.ts

```diff
@@ -1,98 +1,107 @@
-import { Reporter } from './Reporter';
-import { Logger } from './Logger';
-
-const OPTIN_REQUESTED_KEY = 'redhat.telemetry.optInRequested';
+import * as vscode from "vscode";
+import { Logger } from "./utils/logger";
+import { TelemetryEventQueue } from "./utils/telemetryEventQueue";
+import { TelemetryService } from "./services/telemetryService";
 
+// this method is called when your extension is activated
 export function activate(context: vscode.ExtensionContext) {
-  Reporter.initialize(context);
-
-  if (!getTelemetryEnabledConfig()) {
-    Logger.log('redhat.telemetry.enabled is false');
+  Logger.log('"vscode-commons" is now active!');
+
+  /* 
+    check for current status of telemetry true | false
+    initialize events queue to preserve events 
+  */
+  if (!TelemetryService.getTelemetryEnabledConfig()) {
+    Logger.log("redhat.telemetry.enabled is false");
     TelemetryEventQueue.initialize();
   } else {
-    Logger.log('redhat.telemetry.enabled is true');
+    Logger.log("redhat.telemetry.enabled is true");
   }
 
-  openTelemetryOptInDialogIfNeeded(context);
+  TelemetryService.openTelemetryOptInDialogIfNeeded(context);
+
+  context.subscriptions.push(
+    vscode.commands.registerCommand(
+      "vscodeCommons.openWebPage",
+      TelemetryService.openWebPage
+    )
+  );
 
+  /* 
+    listener for configuration change
+  */
   context.subscriptions.push(onDidChangeTelemetryEnabled());
 
-  // This command exists only for testing purposes. Should delete later.
-  context.subscriptions.push(vscode.commands.registerCommand('vscodeCommons.clearStateAndSettings', () => {
-    context.globalState.update('optInRequested', undefined);
-    vscode.workspace.getConfiguration('redhat.telemetry').update('enabled', undefined, true);
-  }));
+  /* 
+    test command to activate and  view telemetry status of 
+    extension through command pallet can be removed later 
+  */
+  context.subscriptions.push(
+    vscode.commands.registerCommand("vscodeCommons.showTelemetryStatus", () => {
+      vscode.window.showInformationMessage(
+        `Red Hat Telemetry Enabled: ${TelemetryService.getTelemetryEnabledConfig()}`
+      );
+    })
+  );
 
-  context.subscriptions.push(vscode.commands.registerCommand('vscodeCommons.openWebPage', openWebPage));
+  // This command exists only for testing purposes. Should delete later.
+  context.subscriptions.push(
+    vscode.commands.registerCommand(
+      "vscodeCommons.clearStateAndSettings",
+      () => {
+        context.globalState.update("OPT_IN_STATUS", undefined);
+        vscode.workspace
+          .getConfiguration("redhat.telemetry")
+          .update("enabled", undefined, true);
+      }
+    )
+  );
+
+  /* 
+    These are the APIs which  are exposed by this extension
+    These APIs can be used in by other extensions as exports
+    Please view INTEGRATION.md for more details
+  */
+  // export to other extensions
+  return Promise.resolve({
+    TelemetryService,
```

Comment:
getTelemetryService

Reply:
replaced it by a getter.

Reply:
> @benoitf what's the minimal supported vscode version in Cheia?

Usually it's following vscode release with like minus one

For now it's 1.50.0
https://github.com/eclipse-theia/theia/blob/master/packages/plugin-ext-vscode/src/node/plugin-vscode-init.ts#L22

Reply:
thanks